From a08565df495036b40d712aa3fdcb1d42e8f3cd4b Mon Sep 17 00:00:00 2001
From: Pawel Osciak <posciak@chromium.org>
Date: Fri, 19 Aug 2016 18:12:09 +0900
Subject: [PATCH] CHROMIUM: v4l: Add
 V4L2_CID_MPEG_VIDEO_H264_SPS_PPS_BEFORE_IDR

When set, the V4L2_CID_MPEG_VIDEO_H264_SPS_PPS_BEFORE_IDR control requests
an H.264 encoder to insert an SPS and PPS pair before each IDR.

Signed-off-by: Pawel Osciak <posciak@chromium.org>

BUG=chromium:639238

[fbuergisser@chromium.org: original patch modified as h264 headers were moved
 to other header files which cause a failure when applying the original patch.
 Keeping part of the original patch for context.]
---
 include/uapi/linux/v4l2-controls.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/include/uapi/linux/v4l2-controls.h b/include/uapi/linux/v4l2-controls.h
index e65de7df0866..830eae8f9d0d 100644
--- a/include/uapi/linux/v4l2-controls.h
+++ b/include/uapi/linux/v4l2-controls.h
@@ -531,6 +531,7 @@ enum v4l2_mpeg_video_h264_hierarchical_coding_type {
 };
 #define V4L2_CID_MPEG_VIDEO_H264_HIERARCHICAL_CODING_LAYER	(V4L2_CID_MPEG_BASE+381)
 #define V4L2_CID_MPEG_VIDEO_H264_HIERARCHICAL_CODING_LAYER_QP	(V4L2_CID_MPEG_BASE+382)
+#define V4L2_CID_MPEG_VIDEO_H264_SPS_PPS_BEFORE_IDR	(V4L2_CID_MPEG_BASE+388)
 #define V4L2_CID_MPEG_VIDEO_MPEG4_I_FRAME_QP	(V4L2_CID_MPEG_BASE+400)
 #define V4L2_CID_MPEG_VIDEO_MPEG4_P_FRAME_QP	(V4L2_CID_MPEG_BASE+401)
 #define V4L2_CID_MPEG_VIDEO_MPEG4_B_FRAME_QP	(V4L2_CID_MPEG_BASE+402)
-- 
2.22.0.657.g960e92d24f-goog

